parser_library/src/workspaces/workspace.cpp

```diff
@@ -411,9 +411,21 @@ lsp::completion_list_s workspace::generate_completion(const lsp::completion_list
 
     const auto& hlasm_ctx = cli.lsp_ctx->get_related_hlasm_context();
 
-    auto suggestions = !cli.completed_text.empty() && instruction_suggestions
-        ? instruction_suggestions(cli.completed_text)
-        : std::vector<std::string>();
+    auto suggestions = [&instruction_suggestions](std::string_view ct) {
+        std::vector<std::pair<std::string, bool>> result;
+        if (ct.empty() || !instruction_suggestions)
+            return std::vector<std::pair<std::string, bool>>();
```

Comment:
Maybe just return `result` since it's already created?

```suggestion
            return result;
```